mergify_engine/web/api/queues.py

```diff
@@ -317,5 +324,435 @@ async def repository_queues(
+            )
+
+
+@router.get(
+    "/repos/{owner}/{repository}/queue/{queue_name}/unpause",  # noqa: FS003
+    summary="Unpause a specific merge queue",
+    description="Unpause the specified merge queue only if there is no paused queue above its priority level",
+    responses={
+        **api.default_responses,  # type: ignore
+        404: {"description": "Not found"},
+        200: {"description": "The queue have been unpaused"},
+    },
+)
+async def unpause_queue(
+    owner: github_types.GitHubLogin = fastapi.Path(  # noqa: B008
+        ..., description="The owner of the repository"
+    ),
+    repository: github_types.GitHubRepositoryName = fastapi.Path(  # noqa: B008
+        ..., description="The name of the repository"
+    ),
+    queue_name: rules.QueueName = fastapi.Path(  # noqa: B008
+        ..., description="The name of the queue"
+    ),
+    redis_cache: utils.RedisCache = fastapi.Depends(  # noqa: B008
+        redis.get_redis_cache
+    ),
+    redis_queue: utils.RedisQueue = fastapi.Depends(  # noqa: B008
+        redis.get_redis_queue
+    ),
+    installation_json: github_types.GitHubInstallation = fastapi.Depends(  # noqa: B008
+        security.get_installation
+    ),
+) -> responses.JSONResponse:
+
+    async with github.aget_client(installation_json) as client:
+        try:
+            # Check this token as access to this repository
+            repo = typing.cast(
+                github_types.GitHubRepository,
+                await client.item(f"/repos/{owner}/{repository}"),
+            )
+        except (http.HTTPNotFound, http.HTTPForbidden, http.HTTPUnauthorized):
+            raise fastapi.HTTPException(status_code=404)
+
+        sub = await subscription.Subscription.get_subscription(
+            redis_cache, repo["owner"]["id"]
+        )
+
+        installation = context.Installation(
+            installation_json, sub, client, redis_cache, redis_queue
+        )
+        repository_ctxt = installation.get_repository_from_github_data(repo)
+        config_file_scheme = await repository_ctxt.get_mergify_config_file()
+
+        if config_file_scheme and installation.redis_queue:
+            config_file = get_mergify_config(config_file_scheme)
+
+            repo_id = repo["id"]
+            owner_id = installation.owner_id
+            queue_rules = config_file["queue_rules"]
+
+            if queue_name not in [rule.name for rule in queue_rules]:
+                return responses.JSONResponse(
+                    status_code=400,
+                    content={"error": f'The queue "{queue_name}" does not exists'},
+                )
+
+            queue_name_priority = queue_rules.get_prority_level(queue_name)
+            redis_hash = f"merge-pause~{owner_id}"
+
+            for rule in queue_rules:
+                redis_key = f"{repo_id}~{rule.name}"
+
+                if queue_name_priority is not None and (
+                    rule.config["priority"] > queue_name_priority
```

Comment:
why are we dealing with priorities here?

Reply:
As I mentionned uper, though I misunderstood "priority" and "process order", it also answer to the induced falling behavior of the pause system, "if we pause the requested queue, and all queues behind in the process order, we don't need to interact with the queues before in the process order". Thus I check the process order to avoid this